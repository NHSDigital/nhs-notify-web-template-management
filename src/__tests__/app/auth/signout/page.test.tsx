import { ReactNode, ComponentType } from 'react';
import MockSignoutPage from '@app/auth/signout/page.dev';
import { render, screen } from '@testing-library/react';

jest.mock('@aws-amplify/auth', () => ({
  signOut: () => Promise.resolve(),
}));

jest.mock('@molecules/Redirect/Redirect', () => ({
  Redirect: () => <p>redirect</p>,
}));

jest.mock('@aws-amplify/ui-react', () => ({
  Authenticator: {
    Provider: ({ children }: { children: ReactNode }) => children,
  },
  withAuthenticator: (Component: ComponentType) => Component,
}));

test('MockSignoutPage', async () => {
  const container = render(<MockSignoutPage />);

  await screen.findByText('redirect');

  expect(container.asFragment()).toMatchSnapshot();
});
