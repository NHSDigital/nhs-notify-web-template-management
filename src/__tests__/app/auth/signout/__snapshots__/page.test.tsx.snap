// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockSignoutPage 1`] = `
<DocumentFragment>
  <p>
    redirect
  </p>
</DocumentFragment>
`;
